docs/decisions/0035-embedding-abstraction.md

```diff
@@ -0,0 +1,257 @@
+---
+# These are optional elements. Feel free to remove any of them.
+status: proposed
+contact: Krzysztof318
+date: 2024-03-02
+deciders: ???
+consulted: ???
+informed: ???
+---
+
+# Embeddings Generation in Semantic Kernel
+
+## Context and Problem Statement
+
+### General Information
+
+The current abstraction of generating embeddings only allows for generation from text data.\
+We want to enable the generation of embeddings from other types of data, such as images and others.\
+We also want to make it possible to parameterize the query.
+
+### Embeddings models
+
+OpenAI provides models that allow for generating embeddings only from textual data.\
+The models do not return metadata, and also do not allow for parameterizing the query.
```

Comment:
It looks like OpenAI API for embeddings does return metadata (token usage) and does allow to pass additional parameters. From documentation:
![image](https://github.com/microsoft/semantic-kernel/assets/13853051/d0bd3487-2c9e-4082-91dd-dc0a684ba681)